metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
Actually, since sensorParserConfig is so nicely cached as a POJO, why pass it through arguments at all?  GrokParser can just fetch it again, for the comparison in GrokParser:134.  Then the signature of parse() can remain the same and none of the client parsers or their unit tests need to change; just the unit test for overall GrokParser.  Am I missing something?